drivers/dma/dma_mcux_edma.c

```diff
@@ -519,54 +629,93 @@ static int dma_mcux_edma_init(const struct device *dev)
 			   irq_enable(DT_INST_IRQ_BY_IDX(n, idx, irq));	     \
 			   ))
 
-#define DMA_MCUX_EDMA_CONFIG_FUNC(n)				      \
-	static void dma_imx_config_func_##n(const struct device *dev) \
-	{							      \
-		ARG_UNUSED(dev);				      \
-								      \
-		IRQ_CONFIG(n, 0, dma_mcux_edma_irq_handler);	      \
-		IRQ_CONFIG(n, 1, dma_mcux_edma_irq_handler);	      \
-		IRQ_CONFIG(n, 2, dma_mcux_edma_irq_handler);	      \
-		IRQ_CONFIG(n, 3, dma_mcux_edma_irq_handler);	      \
-		IRQ_CONFIG(n, 4, dma_mcux_edma_irq_handler);	      \
-		IRQ_CONFIG(n, 5, dma_mcux_edma_irq_handler);	      \
-		IRQ_CONFIG(n, 6, dma_mcux_edma_irq_handler);	      \
-		IRQ_CONFIG(n, 7, dma_mcux_edma_irq_handler);	      \
-		IRQ_CONFIG(n, 8, dma_mcux_edma_irq_handler);	      \
-		IRQ_CONFIG(n, 9, dma_mcux_edma_irq_handler);	      \
-		IRQ_CONFIG(n, 10, dma_mcux_edma_irq_handler);	      \
-		IRQ_CONFIG(n, 11, dma_mcux_edma_irq_handler);	      \
-		IRQ_CONFIG(n, 12, dma_mcux_edma_irq_handler);	      \
-		IRQ_CONFIG(n, 13, dma_mcux_edma_irq_handler);	      \
-		IRQ_CONFIG(n, 14, dma_mcux_edma_irq_handler);	      \
-		IRQ_CONFIG(n, 15, dma_mcux_edma_irq_handler);	      \
-								      \
-		IRQ_CONFIG(n, 16, dma_mcux_edma_error_irq_handler);   \
-								      \
-		LOG_DBG("install irq done");			      \
+#ifdef CONFIG_DMA_MCUX_SINGLE_IRQ_PER_CHANNEL
```

Comment:
It seems like this macro will install an interrupt handler per each channel of the EDMA. Existing NXP SOCs using the legacy EDMA also have a per channel interrupt, but have a combined error interrupt. Would it be possible to enable per channel interrupts for all EDMA instances, and only enable the error interrupt servicing in the channel interrupt when this Kconfig was set?

Reply:
Good idea, it's possible, i just re-pushed the update

Reply:
@danieldegrasse could you pls revisit if this one looks fine to you